src/Behat/Gherkin/Node/ExampleNode.php

```diff
@@ -178,6 +188,30 @@ public function getOutlineTitle()
         return $this->outlineTitle;
     }
 
+    /**
+     * Returns title for this scenario outline example.
+     *
+     * @return string
+     */
+    public function getScenarioTitle()
+    {
+        return "{$this->replaceTextTokens($this->outlineTitle)} #{$this->index}";
+    }
+
+    /**
+     * Returns the example row as a single string.
+     *
+     * You should normally not need this, since it is an implementation detail.
+     * If you need the individual example values, use {@see self::getTokens()}.
+     * To get the fully-normalised/expanded title, use {@see self::getScenarioTitle()}.
+     *
+     * @return string
+     */
+    public function getExampleText()
```

Comment:
There is still at least one valid usecase - for a formatter that wants to print the table row with the exact padding it needs to align with rows above and below.

That said I have done some more digging and it seems Behat currently does this two different ways:
* [PrettyExamplePrinter](https://github.com/Behat/Behat/blob/master/src/Behat/Behat/Output/Node/Printer/Pretty/PrettyExamplePrinter.php#L72) just calls `$example->getTitle()`.
* [PrettyExampleRowPrinter](https://github.com/Behat/Behat/blob/bfccc6549596d048e646754d5b1ef36fd71a3f13/src/Behat/Behat/Output/Node/Printer/Pretty/PrettyExampleRowPrinter.php#L83-L85) uses a more complex implementation that searches for the example in the `OutlineNode` to find the row number, then uses the deprecated `$outline->getExampleTable()` to get to the row content. That would be even more complex if rewritten not to use the deprecated method.

It took a while to understand what these classes are, but it turns out `PrettyExamplePrinter` is used when the `expand: true` option is set, while `PrettyExampleRowPrinter` is used if `expand = false`. 

The `PrettyExampleRowPrinter` has to use a more complex approach because it colours individual cells in the row depending on whether they are referenced in a step that failed (see below). So it needs the original padding for vertical alignment, but it also needs the individual `key => value` pairs. However it could potentially be simplified a bit, particularly with the new `index`/`exampleNumber` property.
![image](https://github.com/user-attachments/assets/c5d91593-7edb-43b3-8ade-fc7985a12ae7)

However, I think that it would be a shame (and obviously a BC break) to force the `PrettyExamplePrinter` (or any similar third-party formatter) to implement complex logic like that in `PrettyExampleRowPrinter` just to get the cell padding that it can currently use from `getTitle()`.

Reply:
Oh, wow.. quite unexpected.

OK then, I'll leave it as is.